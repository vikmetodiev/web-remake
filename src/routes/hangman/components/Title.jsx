import React from 'react';
import styled from 'styled-components';

const Section = styled.section`padding: 20px 0 0;`;

const StyledTitle = styled.ul`
  display: inline-block;
  padding: 0;
  li {
    display: inline-block;
    text-transform: uppercase;
    width: 26px;
    margin-bottom: 15px;
    box-sizing: border-box;
    
  }
  li.titleLetter {
    color:azure
    border-bottom: 1px solid #666;
    margin-right: 4px;
    font-weight: bold;
    font-size: 26px;
    span {
      visibility: hidden;
    }
    span.visible {
      visibility: visible;
    }
    &[data-red='true'] span {
      visibility: visible;
      color: #8e3333;
    }
  }
  &:last-child li:last-child {
    display: none;
  }
`;

const Title = props => {
  const sentence = props.content.split(/[ ,]+/);
  const words = sentence.map(word => word.split(''));

  const letters = words.map((word, i) => (
    <StyledTitle key={i}>
      {word.map((letter, i) => (
        <li
          key={i}
          className="titleLetter"
          data-red={props.missedLetters.includes(letter) ? 'true' : 'false'}
        >
          <span className={props.guesses.includes(letter) ? 'visible' : ''}>
            {letter}
          </span>
        </li>
      ))}
      <li>&nbsp;</li>
    </StyledTitle>
  ));

  return <Section className="title">{letters}</Section>;
};

export default Title;
