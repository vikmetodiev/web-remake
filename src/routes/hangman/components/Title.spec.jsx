import React from 'react';

import toJson from 'enzyme-to-json';
import Title from './Title';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Title', () => {
  it('should render the component', () => {
    const wrapper = shallow(<Title 
      content='yay'
      guesses='abc'
      missedLetters='y'
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
