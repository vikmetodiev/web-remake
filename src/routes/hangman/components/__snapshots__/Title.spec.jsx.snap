// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render the component 1`] = `
<styled.section
  className="title"
>
  <styled.ul
    key="0"
  >
    <li
      className="titleLetter"
      data-red="true"
      key="0"
    >
      <span
        className=""
      >
        y
      </span>
    </li>
    <li
      className="titleLetter"
      data-red="false"
      key="1"
    >
      <span
        className="visible"
      >
        a
      </span>
    </li>
    <li
      className="titleLetter"
      data-red="true"
      key="2"
    >
      <span
        className=""
      >
        y
      </span>
    </li>
    <li>
      Â 
    </li>
  </styled.ul>
</styled.section>
`;
